disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,282 @@
+    """Parse the folder according to the paramerters (data_source,
+    campaign_name and station_name) to get all url from the config files.
+
+    Returns
+    -------
+    list
+        List of tuples containing the local path and the url.
+    """
+    # test if the file path is correct and exists. If not, raise an error
+    check_path(raw_folder)
+
+    # Test that the path ends with DISDRODB/Raw. If not, raise an error
+    split_path = raw_folder.split(os.sep)
+    if split_path[-1] != "Raw" or split_path[-2] != "DISDRODB":
+        raise ValueError("The path must end with DISDRODB/Raw. Please check the path.")
+
+    # Define the path to analyse
+    path_to_analyse = raw_folder
+    if data_source:
+        path_to_analyse = os.path.join(path_to_analyse, data_source)
+    if campaign_name:
+        path_to_analyse = os.path.join(path_to_analyse, campaign_name)
+
+    # Get all config files from the metadata folders
+    metadata_folders = get_metadata_folders(path_to_analyse)
+    yaml_files = get_yaml_files(metadata_folders)
+
+    # Get the list of url and local paths from the config files
+    list_of_path_url = []
+    for yaml_file_path in yaml_files:
+        file_content = yaml.safe_load(open(yaml_file_path))
+        if "data_url" in file_content.keys():
+            data_url = file_content["data_url"]
+            config_station_name = os.path.basename(yaml_file_path).replace(".yml", "")
+            # Only if the station name is not specified or if the station name is the same as the config file name
+            if not station_name or station_name == config_station_name:
+                folder_path = os.path.dirname(yaml_file_path).replace("metadata", "data")
+                file_path = os.path.join(folder_path, config_station_name)
+                list_of_path_url.append((file_path, data_url))
+
+    return list_of_path_url
+
+
+def check_url(url: str) -> bool:
+    """Check url
+
+    Parameters
+    ----------
+    url : str
+        url
+
+    Returns
+    -------
+    bool
+        True if url well formated, False if not well formated
+    """
+    regex = r"^(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)$"  # noqa: E501
+
+    if re.match(regex, url):
+        return True
+    else:
+        return False
+
+
+def download_file_from_url(url: str, folder_path: str) -> None:
+    """Download file
+
+    Parameters
+    ----------
+
+    url : str
+        URL of the file to download
+    file_name : str
+        Local name of the file
+    """
+
+    file_name = os.path.basename(url)
+
+    downloader = pooch.HTTPDownloader(progressbar=True)
+    pooch.retrieve(
+        url=url,
+        known_hash=None,
+        path=folder_path,
+        fname=file_name,
+        downloader=downloader,
+        progressbar=tqdm,
+    )
+
+
+# function to unzip one file into a folder
+def unzip_file(file_path: str, dest_path: str) -> None:
+    """Unzip a file into a folder

```

Comment:
Not very clear. Is a zipped directory here or a zipped file? 
(We don't want to unzip zipped files ... )

Reply:
We want to unzip the station directory

Reply:
no zip anymore